jsk_spot_robot package for spot arm

k-okada opened the pull request:
based on https://github.com/jsk-ros-pkg/jsk_robot/pull/1586

if you need more function, please check 

- #1382
- #1376
- #1412 
- #1414 

see https://github.com/k-okada/jsk_robot/blob/spot_arm/jsk_spot_robot/README.md for more info
@tkmtnt7000 @a-ichikura

sktometometo committed:
[jsk_spot_startup] add jsk_spot_startup package

sktometometo committed:
[jsk_spot_startup] fix the filepath of a rviz config in rviz.launch

sktometometo committed:
[jsk_spot_startup] update the package.xml to fix the version, a maintainer tag, the author tag, and dependencies

sktometometo committed:
[jsk_spot_startup] add Install section, Testing Section to CMakeLists.txt

sktometometo committed:
[jsk_spot_startup] add play.launch and record.launch

sktometometo committed:
[jsk_spot_startup] update record.launch

sktometometo committed:
[jsk_spot_startup] add peripheral.launch to bringup peripheral devices

sktometometo committed:
[jsk_spot_startup] update driver.launch to add required property to spot_ros.py

sktometometo committed:
[jsk_spot_startup] add panoramma image topic to record.launch and add use_compress arg

sktometometo committed:
[jsk_spot_startup] change rviz view to robot centric and add panorama image

sktometometo committed:
[jsk_spot_startup] update rviz config to add base_link trajectory

sktometometo committed:
[jsk_spot_startup] update play.launch

sktometometo committed:
[jsk_spot_startup] update rviz config to visualize battery percentage

sktometometo committed:
[jsk_spot_startup] update play.launch

sktometometo committed:
[jsk_spot_startup] add jsk_spot_bringup.launch

sktometometo committed:
[jsk_spot_startup] update rviz config

sktometometo committed:
[jsk_spot_startup] add resize node for panorama image

sktometometo committed:
[jsk_spot_startup] update rviz.launch and add rviz config

sktometometo committed:
[jsk_spot_startup] update spot_with_mot rviz config

sktometometo committed:
[jsk_spot_startup] add respeaker_node to peripheral.launch

sktometometo committed:
[jsk_spot_startup] removed unused launch

sktometometo committed:
[jsk_spot_startup] add interaction launch

sktometometo committed:
[jsk_spot_startup] fix bugs in interaction.launch

sktometometo committed:
[jsk_spot_startup] add rviz config

sktometometo committed:
[jsk_spot_startup] fix typo

sktometometo committed:
[jsk_spot_startup] update rviz config

sktometometo committed:
[jsk_spot_startup] use english by default to speech_recognition

sktometometo committed:
[jsk_spot_startup] add japanese online speech recgonition

sktometometo committed:
[jsk_spot_startup] fix topic name

sktometometo committed:
[jsk_spot_startup] update twist_mux to add develop cmd_vel input

sktometometo committed:
[jsk_spot_startup] add respawn attribute to spot_ros.py node in driver.launch

sktometometo committed:
[jsk_spot_startup] fix audio topic name in interaction.launchg

sktometometo committed:
[jsk_spot_startup] mv basic launch to launch/include

sktometometo committed:
[jsk_spot_startup] fix jsk_spot_bringup.launch

sktometometo committed:
[jsk_spot_startup] add additional topics to record.launch

sktometometo committed:
[jsk_spot_startup] update rviz config

sktometometo committed:
[jsk_spot_startup] updte rviz config

sktometometo committed:
[jsk_spot_startup] add cmd_vel_smoother

sktometometo committed:
[jsk_spot_startup] add spot_recognition topics to record.launch

tongtybj committed:
[jsk_spot_startup] (#10)

- make rosbag play node as an independent terminal to enable the keyboard input, suahs as pause
- add fisheye panorama image in rviz

mqcmd196 committed:
[jsk_spot_startup] Add options to the files for rosbag and add dependency (#16)

* rename rosbag launch files

* add duration, start etc...

* add xterm to exec_depend for using rosbag play

sktometometo committed:
[jsk_spot_startup] add laptop_battery_monitor.py to driver.launch

sktometometo committed:
[jsk_spot_startup] add laptop battery visualizer to rviz

sktometometo committed:
[jsk_spot_startup] add object detection visualization to rosbag_play

708yamaguchi committed:
[jsk_spot_startup] Add exec_depend packages to jsk_spot_startup

sktometometo committed:
[jsk_spot_startup] add cable_connection\detector.py

sktometometo committed:
[jsk_spot_startup] add front microphone (#27)

sktometometo committed:
[jsk_spot_startup] update rviz config to use raw image

sktometometo committed:
[jsk_spot_startup] update battery scripts and update launch

sktometometo committed:
[jsk_spot_startup] fix battery scripts

sktometometo committed:
[jsk_spot_startup] add launch_robot_state_publisher arg

sktometometo committed:
[jsk_spot_startup] move battery visualization script to driver.launch

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_startup] update package.xml

sktometometo committed:
[jsk_spot_startup] add use_ args

sktometometo committed:
[jsk_spot_startup] update spot rviz

sktometometo committed:
[jsk_spot_startup] fix driver.launch for spot_ros

sktometometo committed:
[jsk_spot_robot] update README and rosinstall

sktometometo committed:
[.travis] add spot_ros to .travis.rosinstall.melodic

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_robot] update rosinstall

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_startup] fix bugs

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_robot] add postfix setup

sktometometo committed:
[jsk_spot_startup] update cable_connection_detector.py

sktometometo committed:
[jsk_spot_robot] update jsk_recognition branch

sktometometo committed:
[jsk_spot_startup] fix bugs of cable_connection_detector

sktometometo committed:
[jsk_spot_startup] changed spot_credential place to be loaded to /var/lib/robot/credentials

sktometometo committed:
[jsk_spot_startup] add ros_speech_recognition node to interaction.launch

tkmtnt7000 committed:
[jsk_spot_startup] add battery temperature warning

tkmtnt7000 committed:
[jsk_spot_robot] fix typo __main_ -> __main__ and change error messages

knorth55 committed:
[jsk_spot_startup] suppress battery warning

sktometometo committed:
[jsk_spot_startup] add plug_server to driver.launch

sktometometo committed:
[jsk_spot_robot] remove audio_video_recorder from jsk_spot_user.rosinstall

708yamaguchi committed:
[spoteus] Add body inverse kinematics to control spot pose in euslisp

mqcmd196 committed:
[spoteus] support body pose z by sending coords

mqcmd196 committed:
[spoteus] support both r p y and coords, still need test

mqcmd196 committed:
[spoteus] fix bugs and support both vec and coords

mqcmd196 committed:
[spoteus] fix log levec

mqcmd196 committed:
[spoteus] support both vector and list

mqcmd196 committed:
[spoteus] support new body pose in sample

sktometometo committed:
[spoteus] Restrict XY movement in spot :fullbody-inverse-kinematics

sktometometo committed:
[jsk_spot_startup] add shinjo to maintainers

sktometometo committed:
[jsk_spot_robot] Update udev rules installation instruction

sktometometo committed:
[spoteus][spoteus_demo] add spoteus_demo package and move demo scripts from spoteus

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_startup] split credential to yaml file

sktometometo committed:
[jsk_spot_robot] update README.md

sktometometo committed:
[jsk_spot_robot] add gdrive_ros to rosinstall

sktometometo committed:
[jsk_spot_robot] update documentation

sktometometo committed:
[jsk_spot_robot] move jsk_spot_user.rosinstall to jsk_spot_dev and add new one

sktometometo committed:
[spoteus] update package.xml for dependencies

mqcmd196 committed:
[jsk_spot_startup] use voice_text as default

tkmtnt7000 committed:
[spoteus] add method for dock and undock

sktometometo committed:
[jsk_spot_startup] update package.xml and CMakeLists.txt

sktometometo committed:
[jsk_spot_startup] add battery perceptage speech

sktometometo committed:
[jsk_spot_startup] fix format for python

sktometometo committed:
[jsk_spot_startup] go back to dock when battery low

tkmtnt7000 committed:
[spoteus] add sample script of :go-to-spot

sktometometo committed:
[jsk_spot_startup] update battery_notifier

sktometometo committed:
[jsk_spot_startup] remove quatered panorama image

sktometometo committed:
[jsk_spot_startup] support audio device settings

sktometometo committed:
[jsk_spot_startup] fix battery notifier remapping

k-okada committed:
spoteus/CMakeLists.txt: use collada2eus to generate lisp from urdf, not dae for Noetic, see https://github.com/ros/collada_urdf/issues/43

k-okada committed:
jsk_spot_robot/requirements.txt: update bosdyn to 3.2.0

k-okada committed:
spoteus: use https://github.com/cst0/spot_ros for arm-spot, update spot.yml, add interface to spot-interface.l

k-okada committed:
update to use https://github.com/cst0/spot_ros, which has better arm interfaces

k-okada committed:
[spoteus/spot-interface.l] support /spot/status/manipulator_state

k-okada committed:
jsk_spot_startup/launch/jsk_spot_bringup.launch: add launch_interaction and launch_peripheral args

k-okada committed:
update jsk_spot_startup/launch/include/driver.launch to support teleop

add teleop node to jsk_spot_robot/jsk_spot_startup/launch/include/driver.launch
update config/twist_mux.yaml for mux head_teleop and joy_teleop
add image compressed node add teleop node to jsk_spot_robot/jsk_spot_startup/launch/include/driver.launch
update package.xml for image compresed node

k-okada committed:
jsk_spot_robot/jsk_spot_driver.rosinstall: add jsk_mode_tools with custom limb support

k-okada committed:
jsk_robot_common/jsk_robot_startup/config/dualsense_joystick_teleop.yaml: # tuck/untuck need 'Unlock button'

k-okada committed:
[jsk_spot_startup] add apps

k-okada committed:
[quadruped_joystick_teleop.cpp] show response messages too

k-okada committed:
[jsk_spot_driver.rosinstall] add robot_upstart, aques_talk, app_manager

k-okada committed:
jsk_spot_startup/{battery_notifier.py, /cable_connection_detector.py] remove laptop battery status

k-okada committed:
jsk_spot_startup/launch/rosbag_record.launch: we only needs compress, if you need raw topics, we do not want to write same thing twice

k-okada committed:
launch/jsk_spot_bringup.launch add robot/type, robot/name, see https://github.com/ros-infrastructure/rep/pull/104

k-okada committed:
[jsk_spot_startup/launch/include/driver.launch] update minimum driver startup file

k-okada committed:
use jsk_spot_startup/config/head_teleop_twist_joy.yaml

k-okada committed:
add node_scripts/call_spot_dock_with_id.py, which support /spot/dock_with_id with Trigger service message

k-okada committed:
jsk_spot_startup/node_scripts/battery_notifier.py: notify battery status every 5 min

k-okada committed:
update jsk_spot_robot/SetupInternalPCAndSpotUser.md, some important information is also removed in this commit, so pleaes check when you improve next time

k-okada committed:
update URL of spot_ros

k-okada committed:
add end_effector_to_joy.py

k-okada committed:
end_effector_to_joy.py: use float for maxrange, add offset_x, offset_y

k-okada committed:
call estop using raw rospy API

k-okada committed:
update head_teleop_joy param

k-okada committed:
support min/max of x,y,yaw

k-okada committed:
jsk_spot_startup/launch/jsk_spot_bringup.launch: enable to pass all args to driver.launch

k-okada committed:
jsk_spot_robot/jsk_spot_driver.rosinstall:fix typo sques_atlk -> aques_talk

k-okada committed:
update README

k-okada committed:
Merge branch 'master' into spot_arm

k-okada committed:
add jsk_spot_user.rosinstall

a-ichikura committed:
fix supervisor and app chooser URLs

k-okada committed:
Merge pull request #81 from a-ichikura/fix_url_typo

fix supervisor and app chooser URLs

k-okada commented:
add mongo setting for spot robot https://gist.github.com/k-okada/74bbfd6af58ce43142375637a8b23252
based on https://github.com/jsk-ros-pkg/jsk_robot/pull/1708

k-okada committed:
add start-grasp/stop-grasp method

k-okada committed:
Merge pull request #83 from k-okada/add_grasp_method

add start-grasp/stop-grasp method

k-okada committed:
add missing dependencse, fix style

k-okada committed:
Merge branch 'spot_arm' of github.com:k-okada/jsk_robot into spot_arm

k-okada committed:
jsk_robot_common/jsk_robot_startup/CMakeLists.txt: install python code using catkin_install_python

k-okada committed:
jsk_robot_startup/lifelog/common_logger.launch add save_faces and save_dialogflow

k-okada committed:
jsk_robot_startup/package.xml: update to version 3, add python3-* packages to exec_depend

k-okada committed:
install orocos_kdl_python, mongodb_store, catkin for compile python3

k-okada committed:
SetupInternalPCAndSpotUser.md: add information for more python3 installs

k-okada committed:
add launch/include/lifelog.launch

k-okada commented:
@a-ichikura  @mqcmd196  @tkmtnt7000 
現状のログ取りシステム
```
ssh strelka -l a-ichikura
tmux a
roscd jsk_spot_startup/; roslaunch  --screen launch/include/lifelog.launch  ;; make-memory の前にこれを立ち上げるのが重要．意識的に`lifelog/face_logger`ノードを立ち上げなおしているので．．．
roscd jsk_spot_startup;  roslaunch --screen launch/make_memory.launch
```
実験する

ログの見かた，73b2-desktop で

```
cd ~/mongo_ws/src/jsk_robot/jsk_robot_common/jsk_robot_startup/scripts
./robot_databse_mongo_dump.py --host 133.11.216.166 --collection StrelKa --start "2022-11-11 18:20:00"
```
これで，2022/11/11 18:20 分以降のデータがダウンロードされてフォルダに格納される．

![Screenshot from 2022-11-11 18-39-40](https://user-images.githubusercontent.com/493276/201312141-9c9abbb3-3506-4721-b72a-25e654503786.png)

- [ ] 表情を認識した瞬間の画像を保存したい．

k-okada committed:
use jsk_spot_startup/config/dualsense_teleop_twist_joy.yaml

k-okada committed:
fix config/dualsense_teleop_twist_joy.yaml to speedup

k-okada committed:
end_effector_to_joy.py: use position, not force for better performance.

k-okada committed:
fix jsk_spot_startup/config/head_teleop_twist_joy.yaml: head_teleop_joy_completion.py did not consider scale_linear, use x_min/x_max etc...

k-okada committed:
use jsk_spot_robot/jsk_spot_startup/CMakeLists.txt: catkin_install_python to install scripts

k-okada committed:
head-lead-teleop.l: run start-func(set-impedance) when R Joy Stick is pressed

k-okada committed:
use jsk_topic_tools/SynchronizedThrottle to generate compressed/thorttle images for mongodb

k-okada committed:
add sample_pick_object_in_image.launch ../node_scripts/sample_pick_object_in_image.l

k-okada committed:
fix head-end-coords : set z-axis to up, to follow jsk style, suppot (send robot :head :look-at), add tests

k-okada committed:
launch/include/driver.launch: fix /spot/status/battery_percentage type to Float32

k-okada committed:
use 'arm' branch of k-okada/spot_ros-arm reposity, instead of hash

k-okada committed:
add *ri* interface to :pick-object-in-image

k-okada committed:
use logwarn for enable/distable center information

k-okada committed:
set respawn true for battery notifier

k-okada committed:
use google_chat ad dialogfow in interaction.launch

k-okada committed:
save data only when /publish_trigger_mongodb_event published

k-okada committed:
add ublox gps driver

k-okada committed:
fix jsk_spot_startup/node_scripts instal location

k-okada committed:
fix config/dualsense_teleop_twist_joy.yaml parameters

k-okada committed:
update head-lead-teleop, add to play with dogtoy

k-okada committed:
add google_chat_logger, fix /aws_auto_checkin_app/class location

k-okada committed:
add perception.launch, ncb_provier, aws_auto_checkin

k-okada committed:
Merge remote-tracking branch 'jsk-ros-pkg/master' into spot_arm

k-okada committed:
add coreio extentions tools

knorth55 commented:
https://github.com/jsk-ros-pkg/jsk_robot/pull/1701/commits/3cc2848f19cc187388a337131c246c2d417acee0

docker-requirements.txt is missing...

https://dev.bostondynamics.com/docs/python/fetch_tutorial/fetch6#

k-okada commented:
sorry.. updated
https://github.com/jsk-ros-pkg/jsk_robot/pull/1701/commits/f82437e235ffddc52b03c0f659970425c2a560f1
--
◉ Kei Okada



2022年12月8日(木) 18:14 Shingo Kitagawa ***@***.***>:

> 3cc2848
> <https://github.com/jsk-ros-pkg/jsk_robot/commit/3cc2848f19cc187388a337131c246c2d417acee0>
>
> docker-requirements.txt is missing...
>
> https://dev.bostondynamics.com/docs/python/fetch_tutorial/fetch6#
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_robot/pull/1701#issuecomment-1342316458>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADYNXCQKBQGM3HARHJNO4TWMGREVANCNFSM6AAAAAARMRANLQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

knorth55 commented on jsk_spot_robot/coreio_extension/ncb_object_detection_coco.service:
```diff
@@ -0,0 +1,18 @@
+[Service]
+Restart=always
+User=root
+ExecStartPre=/bin/bash -c "/usr/bin/docker container inspect object_detection_coco 2> /dev/null || /usr/bin/docker load < /opt/object_detection_coco.tar.gz;"
```
```suggestion
ExecStartPre=/bin/bash -c "/usr/bin/docker image inspect object_detection_coco 2> /dev/null || /usr/bin/docker load < /opt/object_detection_coco.tar.gz;"
```

knorth55 replied:
```suggestion
ExecStartPre=/bin/bash -c "/usr/bin/docker container inspect object_detection_coco:${TAG} 2> /dev/null || /usr/bin/docker load < /opt/object_detection_coco.tar.gz;"
```

knorth55 reviewed (changes requested):
--no-cache option should be set to force update `COPY` `dogtoy-model` section.

knorth55 commented on jsk_spot_robot/coreio_extension/Makefile:
```diff
@@ -0,0 +1,29 @@
+	[ -e models-with-protos ] || unzip models-with-protos.zip
+	[ -e ssd_resnet50_v1_fpn_640x640_coco17_tpu-8.tar.gz ] || wget http://download.tensorflow.org/models/object_detection/tf2/20200711/ssd_resnet50_v1_fpn_640x640_coco17_tpu-8.tar.gz
+	[ -e ssd_resnet50_v1_fpn_640x640_coco17_tpu-8 ] || tar -xvzf ssd_resnet50_v1_fpn_640x640_coco17_tpu-8.tar.gz
+	[ -e dogtoy-model ] || cp -r ../dogtoy/exported-models/dogtoy-model .
+	[ -e dogtoy-model/label_map.pbtxt ] || cp ../dogtoy/annotations/label_map.pbtxt dogtoy-model/
```
```suggestion
        [ ! -e dogtoy-model] || rm dogtoy-model
	cp -r ../dogtoy/exported-models/dogtoy-model .
	cp ../dogtoy/annotations/label_map.pbtxt dogtoy-model/
```

knorth55 replied:
force updating the model

knorth55 commented on jsk_spot_robot/coreio_extension/ncb_object_detection_coco.service:
```diff
@@ -0,0 +1,18 @@
+
+[Service]
+Restart=always
+User=root
```
```suggestion
User=root
Environment=TAG=latest
```

knorth55 commented on jsk_spot_robot/coreio_extension/ncb_object_detection_coco.service:
```diff
@@ -0,0 +1,18 @@
+ExecStartPre=/bin/bash -c "/usr/bin/docker container inspect object_detection_coco 2> /dev/null || /usr/bin/docker load < /opt/object_detection_coco.tar.gz;"
+ExecStartPre=-/usr/bin/docker stop object_detection_coco
+ExecStartPre=-/usr/bin/docker rm object_detection_coco
+ExecStart=/usr/bin/docker run --name object_detection_coco --network host --rm -v /opt/payload_credentials/payload_guid_and_secret:/opt/payload_credentials/payload_guid_and_secret object_detection_coco -m /data/dogtoy-model/saved_model /data/dogtoy-model/label_map.pbtxt -m /data/ssd_resnet50_v1_fpn_640x640_coco17_tpu-8/saved_model /data/ssd_resnet50_v1_fpn_640x640_coco17_tpu-8/mscoco_label_map.pb ...
```
```suggestion
ExecStart=/usr/bin/docker run --name object_detection_coco --network host --rm -v /opt/payload_credentials/payload_guid_and_secret:/opt/payload_credentials/payload_guid_and_secret object_detection_coco:${TAG} -m /data/dogtoy-model/saved_model /data/dogtoy-model/label_map.pbtxt -m /data/ssd_resnet50_v1_fpn_640x640_coco17_tpu-8/saved_model /data/ssd_resnet50_v1_fpn_640x640_coco17_tpu-8/mscoco_label_map.pbtxt --payload-credentials-file /opt/payload_credentials/payload_guid_and_secret --name ssd-resnet-coco-server --port 50055 192.168.50.3
```

k-okada committed:
add ublox in rosbag record

k-okada committed:
update head-lead-teleop.l for ichikura demo

knorth55 commented:
@k-okada 

@a-ichikura さんのデモのコードをBelkaの体内からPRにしました．
マージお願いします．
https://github.com/k-okada/jsk_robot/pull/88

また，こちらのE Stopを間違えて押してしまう問題について，同時押しにしたPRもマージお願いします．
https://github.com/k-okada/jsk_robot/pull/84

knorth55 commented:
I also update the detection model to strelka, too.
![Screenshot_20221216-174012](https://user-images.githubusercontent.com/9300063/208062077-4ad8371e-1cbe-47f7-bf1d-97901125e7f1.png)

knorth55 committed:
add allow-other-keys for args

k-okada committed:
fix typo on jsk_spot_driver.rosinstall/ubox

k-okada committed:
Merge pull request #89 from knorth55/spot_arm-interface

add allow-other-keys for args

knorth55 commented on jsk_spot_robot/jsk_spot_startup/launch/include/driver.launch:
```diff
@@ -0,0 +1,170 @@
+  <!-- joystick teleop nodes -->
+  <include file="$(find jsk_robot_startup)/launch/quadruped_joystick_teleop.launch" >
+    <arg name="pad_type" value="dualsense" />
+    <arg name="joy_dev" value="/dev/input/js0" />
```
it is better to write udev file as below and use `/dev/dualsense` for device path.
otherwise, the first joy device will be used for teleoperation.
Belka has this internal change to use different joystick and dualsense

```
KERNEL=="js*", ATTRS{name}=="Wireless Controller", SYMLINK+="dualsense"
```

k-okada committed:
add use_gps

k-okada committed:
jsk_spot_robot/coreio/base: add development environment for CoreI/O, see https://dev.bostondynamics.com/docs/payload/coreio_documentation#developing-on-the-core-i-o, it is recommended to build on docker file....

k-okada committed:
Merge branch 'spot_arm' of https://github.com/k-okada/jsk_robot into spot_arm

k-okada committed:
add script to buid network compute server docekr images, see https://dev.bostondynamics.com/docs/python/fetch_tutorial/fetch3

k-okada committed:
network_compute_server.py: fix network_compute_server.py for core i/o, see https://dev.bostondynamics.com/docs/python/fetch_tutorial/fetch6

k-okada committed:
update Documents for CORE I/O

k-okada committed:
coreio/base/Dockerfile: set TimeZone

k-okada committed:
coreio/base/Makefile: create and share ~/.ros directory

k-okada committed:
[jsk_spot_startup] unstow arm in every cases

k-okada committed:
Update mongodb_replication_params.yaml

`robot-database.jsk.imi.i.u-tokyo.ac.jp` is CNAME of `musca.jsk....` 
maybe this is the cause of following error
```
    [/message_store /opt/ros/melodic/lib/python2.7/dist-packages/rospy/topics.py:753] [ERROR] [1681119907.337053]: bad callback: <bound method MessageStore.insert_ros_msg of <__main__.M
essageStore object at 0x7f750d1860>>                                                                                                                                                 Traceback (most recent call last):                                                                                                                                                   
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rospy/topics.py", line 750, in _invoke_callback                                                                                     cb(msg)                                                                                 
  File "/home/spot/ws/src/mongodb_store/mongodb_store/scripts/message_store_node.py", line 103, in insert_ros_msg                                                                        self.insert_ros_srv(msg)                                                                                                                                                         
  File "/home/spot/ws/src/mongodb_store/mongodb_store/scripts/message_store_node.py", line 152, in insert_ros_srv                                                                        dc_util.store_message(extra_collection, obj, meta, obj_id)                                                                                                                       
  File "/home/spot/ws/src/mongodb_store/mongodb_store/src/mongodb_store/util.py", line 258, in store_message                                                                             return collection.insert(doc)                                                                                                                                                    
  File "/usr/lib/python3/dist-packages/pymongo/collection.py", line 2941, in insert                                                                                                      check_keys, manipulate, write_concern)                                                                                                                                           
  File "/usr/lib/python3/dist-packages/pymongo/collection.py", line 599, in _insert                                                                                                      bypass_doc_val, session)                                                                                                                                                         
  File "/usr/lib/python3/dist-packages/pymongo/collection.py", line 580, in _insert_one                                                                                                  _check_write_command_response(result)                                                       
  File "/usr/lib/python3/dist-packages/pymongo/helpers.py", line 207, in _check_write_command_response                                                                               
    _raise_last_write_error(write_errors)                                                                                                                                              File "/usr/lib/python3/dist-packages/pymongo/helpers.py", line 188, in _raise_last_write_error                                                                                     
    raise DuplicateKeyError(error.get("errmsg"), 11000, error)                                                                                                                       pymongo.errors.DuplicateKeyError: E11000 duplicate key error collection: jsk_robot_lifelog.strelka index: _id_ dup key: { : ObjectId('6433daa31cf4582bb939adf7') }                   
                                  
```

k-okada committed:
enable to query from extra server (replicate server), keep google_chat_ros data to mongodb

k-okada committed:
add google cloud language and gdrive_ros, update credentials for strelka

k-okada committed:
Merge pull request #93 from sktometometo/PR/spot_arm/update_internal_changes

Pr/spot arm/update internal changes

k-okada committed:
temp

k-okada committed:
picks google cloud arg name

k-okada committed:
Merge pull request #100 from a-ichikura/temp_20230418_internal_interaction

Temp 20230418 internal interaction

sktometometo committed:
Add R1 teleop

sktometometo committed:
[jsk_spot_startup] add switchbot

keitayoneda commented:
kinematic_simulator上で起動する際に`send *ri* :power-on`すると`Failed to establish connection to service server`というエラーが出て起動できなかった
今はspot-interface.lのL526あたりのwhile文をコメントアウトして起動するようにしている

keitayoneda commented:
spot-initを一行ずつ実行していったところ、spot-interfaceのコンストラクタ(:init)で以下のエラーが出た
![Screenshot from 2023-11-15 17-00-32](https://github.com/jsk-ros-pkg/jsk_robot/assets/49116073/784511a9-084f-471e-a3fb-07c583883e2e)

別にターミナルを立ち上げてsourceしたあとに指示通り

```sh
roslaunch spot_driver driver.launch
```

してみると、

```sh
ModuleNotFoundError: No module named 'bosdyn`
```

と言われたので[公式のページ](https://dev.bostondynamics.com/docs/python/quickstart)を参考に

```sh
python3 -m pip install --upgrade bosdyn-client bosdyn-mission bosdyn-choreography-client
```

としてやり直したところ今度は

```sh
ModuleNotFoundError: No module named 'vision-msgs`
```

と言われたので

```sh
sudo apt install ros-noetic-vision-msgs
```

でvision-msgsを入れた。その再びroslaunchすると今度は以下のようにエラーがでた。これはこのlaunchファイルが実機対応？のものなためだと考えられる。
![Screenshot from 2023-11-15 17-29-16](https://github.com/jsk-ros-pkg/jsk_robot/assets/49116073/1ee2bf78-0dc9-4892-b3e6-811c86b977c5)

そこで他のlaunchファイルを探したところ`driver_moch.launch`というものがあったので先程のlaunchのコマンドをこれに書き換えて

```sh
roslaunch spot_driver driver_moch.launch
```

としたところエラーが出なくなった

keitayoneda committed:
[spoteus] remove error in spot-init with kinematics simulator

k-okada committed:
[jsk_spot_robot] spoteus/test/test-spot.l enable test-spot-init

k-okada committed:
Merge pull request #104 from k-okada/spot_arm_sim

remove error in spot-init with kinematics simulator

pazeshun commented on jsk_spot_robot/spoteus/test/test-spot.l:
```diff
@@ -30,8 +31,8 @@
     (look-at robot)))
 
 ;; spot-interface is not work on simulator
```
Perhaps we should remove this comment.

k-okada replied:
thank you. removed.

k-okada committed:
Merge branch 'master' into spot_arm

k-okada committed:
remove outdated comments

k-okada committed:
Merge pull request #109 from sktometometo/PR/spot_arm/add-R1-teleop

Add R1 teleop

k-okada committed:
Merge pull request #110 from sktometometo/PR/spot_arm/add-switchbot-client-ros

[jsk_spot_startup] add switchbot

k-okada committed:
[spoteus] remove find_package(collada_urdf) from CMakeLists.txt

k-okada committed:
add launch/sample_walk_to_point_in_image.launch node_scripts/sample_walk_to_point_in_image.py

k-okada committed:
CMakeLists.txt add install command to work on
> catkin config --install
> catkin build -vi --summarize

we checked that necessary files are installed

```
(export ROS_PACKAGE_PATH=$(pwd)/src; for dir in install/share/*; do pkg=$(basename $dir); echo -e "\n\n$pkg"; echo diff -r install/share/$pkg $(rospack find $pkg) --exclude CMakeLists.txt --exclude .gitignore --exclude setup.py --exclude README.md --exclude CHANGELOG.rst --exclude cmake ; ./check.py $(basename $dir); done > diff.dir)
```
 where `check.py` is

```

import sys
import os
import copy
import filecmp
import rospkg

def compare_directories(install_dir, src_dir, ignore=None):
    # Initialize dictionaries to store results
    results = {
        'unique_to_dir1': [],
        'unique_to_dir2': [],
        'differences': [],
        'common_files': []
    }

    # Recursively compare directory structures and files
    comparison = filecmp.dircmp(install_dir, src_dir, ignore=ignore)

    # Files only in dir1
    results['unique_to_dir1'] = comparison.left_only

    # Files only in dir2
    results['unique_to_dir2'] = comparison.right_only

    # Files present in both directories but different
    results['differences'] = comparison.diff_files

    # Files that are identical in both directories
    results['common_files'] = comparison.same_files

    return results

def list_all_subdirectories(root_dir):
    subdirectories = []
    for dirpath, dirnames, _ in os.walk(root_dir):
        for dirname in dirnames:
            subdirectories.append(os.path.join(dirpath, dirname))
    return subdirectories

if __name__ == '__main__':
    rospack = rospkg.RosPack()
    pkg_name = sys.argv[1]
    pkg_path = rospack.get_path(pkg_name)
    print("## Check [{}]".format(pkg_name))
    for dir in ['.'] + [os.path.relpath(dir, pkg_path) for dir in list_all_subdirectories(pkg_path)]:
        install_dir = os.path.join('install/share/', pkg_name, dir)
        src_dir = os.path.join(pkg_path, dir)
        if os.path.exists(install_dir):
            differences = compare_directories(install_dir, src_dir,
                                              ignore=['env-hooks', 'cmake', '.gitignore', 'CMakeLists.txt', 'CHANGELOG.rst', 'README.md', 'setup.py'])
        else:
            print('> Only in {}'.format(src_dir))
            continue

        # Display the results
        for file in differences['unique_to_dir1']:
            print("> Files only in directory 1: {}".format([os.path.join(dir, file)]))
        for file in copy.copy(differences['unique_to_dir2']):
            if os.path.exists(os.path.join('install/lib/', pkg_name, file)):
                differences['unique_to_dir2'].remove(file)
        for file in differences['unique_to_dir2']:
            print("> Files only in directory 2: {}".format([os.path.join(dir, file)]))
        for file in differences['differences']:
            print("> Different files: {}".format([os.path.join(dir, file)]))
        #print("Common files (identical):", differences['common_files'])

```

```

k-okada committed:
[coreio/base] update docker container for two-stage system, one for spot user to run system service and another one is for development users

k-okada committed:
[jsk_spot_driver.rosinstall]: comment outed aques_talk due to missing arm64 library

k-okada committed:
[jsk_spot_robot/SetupInternalPCAndSpotUser.md] update doc

k-okada committed:
[jsk_spot_robot] fix CMakeLists.txt to work on install layout

k-okada committed:
[jsk_spot_robot/coreio/base/Dockerfile] add example to install debian package for urer docker container

k-okada committed:
[jsk_spot_robot/coreio/base/Makefile] add  catkin build --summarize

k-okada committed:
jsk_spot_robot/coreio/bas/Dockerfile: install pip under system space using 'sudo -H pip3'

k-okada committed:
jsk_spot_startup/launch/jsk_spot_bringup.launch: add launch_switchbot_ros

k-okada committed:
jsk_spot_startup/CMakeLists.txt: add apps to install(DIRECTORY

k-okada committed:
jsk_spot_robot/coreio/base/Makefile: run 'catkin config' with '--install'

k-okada committed:
**FIXME** jsk_spot_robot/coreio/base/Makefile: use --device=/dev/input/js0:/dev/dualsense to run docker

k-okada committed:
jsk_spot_robot/jsk_spot_driver.rosinstall: update spot_ros to support 'Add world object publisher' https://github.com/k-okada/spot_ros-arm/pull/8

k-okada committed:
jsk_spot_robot/coreio/base/Dockerfile: install tmux/expect

k-okada committed:
jsk_spot_robot/coreio/base/Makefile: make shell to support ARGS and NAME

k-okada committed:
jsk_spot_robot/coreio/base/{Makefile,Dockerfile} update for user_build

k-okada committed:
jsk_spot_robot: update documents

mqcmd196 committed:
[jsk_spot_robot] delete git repos which have been released

k-okada committed:
jsk_robot_startup/src/quadruped_joystick_teleop.cpp: say something when service call failed

mqcmd196 committed:
[strelka] enable launch audio related nodes

mqcmd196 committed:
[strelka] add option to launch sound_play_jp

k-okada committed:
Merge pull request #115 from mqcmd196/coreio/enable-audio

[Strelka][coreio] add option to launch sound_play_jp

mqcmd196 committed:
[jsk_spot_startup] add roslist depend

k-okada committed:
Merge pull request #116 from mqcmd196/spot-add-rostest-depend

[jsk_spot_startup] add roslint depend

mqcmd196 committed:
[spot] publish compressed hand_depth_in_hand_color_frame topic

k-okada committed:
Merge pull request #117 from mqcmd196/publish-compressed-hand-depth

[spot] publish compressed hand_depth_in_hand_color_frame topic

k-okada committed:
Merge pull request #120 from mqcmd196/PR/remove-unnecessary-repos

[jsk_spot_robot] remove unnecessary repos

mqcmd196 committed:
[spoteus] add carry mode for arm. requires https://github.com/k-okada/spot_ros-arm/pull/16

k-okada committed:
Merge pull request #122 from mqcmd196/PR/add-carry

[spoteus] add carry mode for arm

mqcmd196 committed:
[jsk_spot_startup] enable voicevox for aarch64 Spot Core I/O

mqcmd196 committed:
[jsk_spot_startup] specify voicevox playing device

mqcmd196 committed:
[spoteus] impl constrained manipulation methods

k-okada committed:
Merge pull request #124 from mqcmd196/add-constrained-manip

[spoteus] impl constrained manipulation methods

mqcmd196 committed:
[spoteus] impl gripping xyz-axis constraint and grasp_palm_to_fingertip in PickObjectInImage client

k-okada committed:
Merge pull request #125 from mqcmd196/add-pick-object-xyz-and-palm

[spoteus] impl gripping xyz-axis constraint and grasp_palm_to_fingertip in PickObjectInImage client

mqcmd196 committed:
[jsk_spot_startup] add voice_text parameters for ReadSpeaker API in strelka

k-okada committed:
Merge pull request #126 from mqcmd196/strelka-enable-ja

[jsk_spot_startup][core-io] Add option to launch voicevox or voice_text

mqcmd196 committed:
[spoteus] add cartesian arm control interface

k-okada committed:
Merge pull request #128 from mqcmd196/PR/add-spot-cartesian-arm

[spoteus] add cartesian arm control interface

mqcmd196 committed:
[spoteus] add grasp holding methods

mqcmd196 committed:
[spoteus] Add claim override (hijack in the tablet)

k-okada committed:
Merge pull request #130 from mqcmd196/override-claim-state

[spoteus] Add claim override (hijack in the tablet)

k-okada committed:
Merge pull request #129 from mqcmd196/add-override-holding

[spoteus] add grasp holding methods
